lib/rules/no-unused-vars.js

```diff
@@ -632,21 +598,26 @@ module.exports = {
                 for (let i = 0, l = unusedVars.length; i < l; ++i) {
                     const unusedVar = unusedVars[i];
 
-                    if (unusedVar.eslintExplicitGlobal) {
-                        context.report({
-                            node: programNode,
-                            loc: getLocation(unusedVar),
-                            message: getDefinedMessage(unusedVar),
-                            data: unusedVar
-                        });
-                    } else if (unusedVar.defs.length > 0) {
+                    // Report the first declaration.
+                    if (unusedVar.defs.length > 0) {
                         context.report({
                             node: unusedVar.identifiers[0],
                             message: unusedVar.references.some(ref => ref.isWrite())
                                 ? getAssignedMessage()
                                 : getDefinedMessage(unusedVar),
                             data: unusedVar
                         });
+
+                    // If there are no regular declaration, report the first `/*globals*/` comment directive.
+                    } else if (unusedVar.eslintExplicitGlobalComments && unusedVar.eslintExplicitGlobalComments.length > 0) {
```

Comment:
To clarify how this works, will `eslintExplicitGlobalComments` always be nonempty if the property exists?

Reply:
You are right. Indeed, this comparison is redundant.